### PR TITLE
Remove usage of Addon._latest_version to prepare for the removal of the field

### PR DESCRIPTION
**Pull request by diox**
It's still set as normal, just no longer used by the `latest_version` property, which is now a `cached_property` to alleviate perf issues.

Part of #3751

### CODE REVIEW

File: src/olympia/addons/models.py [hunk 1]
```diff
@@ -672,17 +672,41 @@ def get_version(self):
...
             fltr = {'files__status__in': status}
-            return self.versions.no_cache().filter(**fltr).extra(
+            version = self.versions.no_cache().filter(**fltr).extra(
                 where=["""
                     NOT EXISTS (
                         SELECT 1 FROM files AS f2
                         WHERE f2.version_id = versions.id AND
                               f2.status NOT IN (%s))
                     """ % status_list])[0]
+            version.addon = self
```
Comment (eviljeff) [inline src/olympia/addons/models.py]: Is this necessary?  Seems dangerous.

File: src/olympia/versions/models.py [hunk 2]
```diff
@@ -576,6 +576,8 @@ def watch_source(old_attr=None, new_attr=None, instance=None, sender=None,
 def update_status(sender, instance, **kw):
```
Comment (eviljeff) [inline src/olympia/versions/models.py]: I assume this is called on version creation somewhere along the line?

**Comment by diox** [comment]
There were initially a ton of tests broken, I had to refine it several time to end up with this, I want to take it very slow. That's why there is no `channel` awareness yet.

File: src/olympia/versions/models.py (hunk 2 above)
Comment (diox) [inline src/olympia/versions/models.py]: Yes, it's called on `post_save` and `post_delete`. I had to clear the cached property and reload the add-on in there to make sure it wasn't overwriting something another signal modified (some tests broke without)

**Pull request merged**